Adding new fields to YAML object to use a different user when doing

BT-rmartin opened the pull request:
querys and to extend the existing context.
Replace some messages printed by dialogs that are raised
Changed structure of XML file, so that it has a container for each
object within /data/object.
Now we can export XML for several items previously chosen or all of them
setting in domain of YAML object [].
Fixed bug with language so that now you can use the field language in
jasper document and set there the language of somo item of the object,
or directly the string with the code of the language.

Commit:
Adding new fields to YAML object to use a different user when doing
querys and to extend the existing context.
Replace some messages printed by dialogs that are raised
Changed structure of XML file, so that it has a container for each
object within /data/object.
Now we can export XML for several items previously chosen or all of them
setting in domain of YAML object [].
Fixed bug with language so that now you can use the field language in
jasper document and set there the language of somo item of the object,
or directly the string with the code of the language.

bt-admin commented on jasper_server/jasper_yaml_object.py:
```diff
@@ -33,5 +33,7 @@ class jasper_yaml_object(osv.osv):
         'offset': fields.integer('Offset'),
         'limit': fields.integer('Limit'),
         'order': fields.char('Order', size=128),
+        'yaml_context': fields.char('Context',),
```
offset and limit are without yaml_prefix. Please keep consistent with that notation.

bt-admin commented on jasper_server/obj_server.py:
```diff
@@ -249,20 +250,31 @@ def generatorYAML(self, cr, uid, jasper_document, current_object, user_company,
+            if yaml_object.yaml_context:
+                yaml_context = ast.literal_eval( yaml_object.yaml_context)
+                for key,value in yaml_context.items():
+                    context[key] = value
```
Please copy the context always before using it. Otherwise you change the original context

bt-admin commented on jasper_server/report/report_soap.py:
```diff
@@ -255,7 +259,8 @@ def _jasper_execute(self, ex, current_document, js_conf, pdf_list,
         if current_document.lang:
             language = self._eval_lang(cur_obj, current_document)
-            language = language[0][0] 
+            if type(language) is not str:  # type is str if language is given directly as string 'de_DE' for instance
```
Better check if language is a list. Then the next language[0] wil
l work

Commit:
Changes required in review. 
Changing names of new fields to be coherent with old ones. 
Copying context before changing it not to override original context.
Changing checking conditions of language

Commit:
Changes to have a tab with Errors in Jasper Document in which we can
check the stack trace and meaningful error messages

Commit:
Removing import that is not needed

Commit:
Fixing some problems when raising errors

BT-jmichaud committed:
Merge pull request #1 from BT-astauder/mara1_improvements_xml_yaml_object

Fixing some problems when raising errors

Commit:
Fix issue in xmlField with object.name

Enable to use it with model without name or rec_name.

Commit:
Merge branch 'mara1_improvements_xml_yaml_object' of ssh://git@github.com/BT-jmichaud/openerp-jasperserver.git into mara1_improvements_xml_yaml_object

BT-astauder committed:
Merge pull request #8 from BT-jmichaud/mara1_improvements_xml_yaml_object

Fix issue in xmlField with object.name

Commit:
Fixed bug when there is no name, and record_name has no value

Commit:
Fixing problem when duplicating jasper document, as it didn't copy the
YAML, parameters and labels

Commit:
Adding anme and value attributes to Jasper Exceptions so that they match
the python exceptions and the same raising attributes can be used for
any kind of exception

Commit:
Merge pull request #11 from brain-tec/8.0

8.0
